modules/vm/main.tf

```diff
@@ -131,3 +132,30 @@ resource "azurerm_linux_virtual_machine_scale_set" "graphdb" {
 
   tags = var.tags
 }
+
+resource "azurerm_role_definition" "backup_role" {
+  name        = "ReadOnlyVMSSStorageRole"
+  scope       = data.azurerm_resource_group.graphdb.id
+  description = "This is a custom role created via Terraform required for creating backups in GraphDB"
+
+  permissions {
+    actions = [
+      "Microsoft.Compute/virtualMachineScaleSets/read",
+      "Microsoft.Storage/storageAccounts/read"
+    ]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    data.azurerm_resource_group.graphdb.id
+  ]
+  depends_on = [data.azurerm_resource_group.graphdb]
```

Comment:
This `depends_on` is not needed because this module already depends on the group in the main.tf.
However, I think that the scale set needs a `depends_on` on this role.

Reply:
sure